src/casync.c

```diff
@@ -1485,6 +1485,8 @@ static int ca_sync_start(CaSync *s) {
                 if (r < 0)
                         return r;
 
+                (void) ca_store_set_compression_type(s->cache_store, s->compression_type);
```

Comment:
this should be unnecessary. Caches don#t recompress, they store everything as it is, i.e. compressed in the compressor of choice, or uncompressed depending on its current state. As such there's no need to propagate the compression type down to the cache store (but it doesn't hurt either)

Reply:
OK. I don't quite understand this part of the code. The is that the cache was failing with ENOSYS for xz, when compiled without xz. If you have a nicer way to make the tests pass, I'm happy to change this.

Reply:
hmm, that indicates a bug I figure. Let's merge it as is then and I'll fix this up later.